huff_codegen/src/lib.rs

```diff
@@ -280,6 +280,26 @@ impl Codegen {
                     bytes.push((starting_offset, Bytes(push_bytes)));
                 }
                 IRByteType::Statement(s) => {
+                    // if we have a codesize call for the constructor here, from within the
+                    // constructor, we skip
```

Comment:
hmm- this is a way to prevent the recursive error, but produces an unexpected output. Ideally, if we don't allow the `__codesize` builtin to be called on the constructor within the constructor, it should throw an error rather than skipping over it. We could definitely go the route of adding a placeholder and filling it at a later point as well, instead of just patching the problem.

For example:
```
#define macro MAIN() = {
    0x00
}

#define macro CONSTRUCTOR() = {
    __codesize(CONSTRUCTOR)
    0x00 mstore
}
```
Produces the creation code:
```
[00]	PUSH1 0x00
[02]	MSTORE	
[03]	PUSH1 0x02
[05]	DUP1	
[06]	PUSH1 0x0c
[08]	RETURNDATASIZE	
[09]	CODECOPY	
[0a]	RETURNDATASIZE	
[0b]	RETURN	
[0c]	PUSH1 00
```
Which causes a stack underflow on deployment.

Reply:
should be fixed 🤝